webapp/javascript/components/TimelineChart/Annotation.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { Maybe } from 'true-myth';
+import type { ExploreTooltipProps } from './ExploreTooltip';
+
+// TODO(eh-am): what are these units?
+const THRESHOLD = 10000;
+
+// TODO: fix types
+export default function Annotations(
+  props: ExploreTooltipProps & {
+    annotations: { timestamp: number; content: string }[];
+  }
+) {
+  if (!props.annotations?.length) {
+    return null;
+  }
+  return getClosestTimestamp(props.values)
+    .andThen((closest) => getClosestAnnotation(props.annotations, closest))
+    .map((annotation) => (
+      <AnnotationComponent
+        timestamp={annotation.timestamp}
+        content={annotation.content}
+      />
+    ))
+    .unwrapOr(null);
+}
+
+function AnnotationComponent({
+  timestamp,
+  content,
+}: {
+  timestamp: number;
+  content: string;
+}) {
+  return (
+    <div>
+      <div>timestamp: {timestamp}</div>
+      <div>content: {content}</div>
+    </div>
+  );
+}
+
+function getClosestTimestamp(
+  values: ExploreTooltipProps['values']
+): Maybe<number> {
+  if (!values) {
+    return Maybe.nothing();
+  }
+  if (!values[0]) {
+    return Maybe.nothing();
+  }
+  if (!values[0].closest) {
+    return Maybe.nothing();
+  }
+  if (!values[0].closest[0]) {
+    return Maybe.nothing();
+  }
+  return Maybe.of(values[0].closest[0]);
+}
+
+function getClosestAnnotation(
+  annotations: { timestamp: number; content: string }[],
+  timestamp: number
+) {
+  const f = annotations.find(
+    (a) => Math.abs(a.timestamp - timestamp) < THRESHOLD
```

Comment:
as I understand you've added `THRESHOLD` to have kind of distance of watching markings, but does it the same for all markings? how does it work you have 3 annotations near each other? for example, on timestamps 15:30:25,  15:30:27, 15:30:28?

Reply:
used a solution, although not perfect, hopefully will fix in https://github.com/pyroscope-io/pyroscope/issues/1505

Reply:
@eh-am, this problem might be a bit deeper, I have no idea why, but these annotation marks (https://github.com/pyroscope-io/pyroscope/issues/1505) looks like vertical selection lines in left/right timelines on ComparisonView page. Just take look at them more attentively. Dotted line + black draggable area. I added this appearance only for left/right single timeline selection lines, is there chance you pass some incorrect props? any kind of props conflict? But no idea what they are doing in annotations PR.
![image](https://user-images.githubusercontent.com/7372044/190429251-49b467df-af60-4cfd-9c57-310da98734df.png)